docs/markdown/guides/mermaid/index.ts

```diff
@@ -0,0 +1,62 @@
+import {DocsCodeToken} from '../extensions/docs-code/docs-code';
+import {chromium} from 'playwright-core';
+import {createRequire} from 'node:module';
+import {Mermaid, MermaidConfig} from 'mermaid';
+import {runfiles} from '@bazel/runfiles';
+import {join} from 'node:path';
+import {mermaidStyling} from './styling';
+
+// Declare mermarid in the context of this file so that typescript doesn't get upset when we
+// access it within the `page.evaluate` function. At runtime the context in with the method
+// is run difference than this file, but this makes typescript happy.
+declare const mermaid: Mermaid;
+
+/** Mermaid configuration to use when creating mermaid svgs. */
+const mermaidConfig: MermaidConfig = {
+  ...mermaidStyling,
+};
+
+/** The full path to the mermaid script. */
+let mermaidScriptTagData: {path: string} | undefined;
+
+/** Get the mermaid script file path, resolving it if necessary first. */
+function getMermaidScriptTagData() {
+  if (mermaidScriptTagData) {
+    return mermaidScriptTagData;
+  }
+  const require = createRequire(import.meta.url);
+  return (mermaidScriptTagData = {path: require.resolve('mermaid/dist/mermaid.js')});
```

Comment:
Ideally resolve via `runfiles` as well as this is a CJS/ESM interop we shouldn't be introducing I think